File: rb/features.rb
# Trivia SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module TriviaFeatures
  def self.make_feature(name)
    case name
    when "base"
      TriviaBaseFeature.new
    when "ratelimit"
      TriviaRatelimitFeature.new
    when "retry"
      TriviaRetryFeature.new
    when "test"
      TriviaTestFeature.new
    when "timeout"
      TriviaTimeoutFeature.new
    else
      TriviaBaseFeature.new
    end
  end
end

File: py/features.py
# Trivia SDK feature factory

from feature.base_feature import TriviaBaseFeature
from feature.test_feature import TriviaTestFeature


def _make_feature(name):
    features = {
        "base": lambda: TriviaBaseFeature(),
        "test": lambda: TriviaTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
